src/core_plugins/kibana/index.js

```diff
@@ -148,6 +149,17 @@ module.exports = function (kibana) {
 
       server.expose('systemApi', systemApi);
       server.expose('handleEsError', handleEsError);
-    }
+    },
+
+    deprecations: function () {
+      return [
+        (settings, log) => {
+          if (has(settings, 'defaultAppId')) {
+            const newConfig = `server.defaultRoute: /app/kibana#/${get(settings, 'defaultAppId')}`;
+            log(`Config key "defaultAppId" will be deprecated. Use the equivalent "${newConfig}"`);
```

Comment:
thoughts on overriding defaultRoute here to `/app/kibana#/${get(settings, 'defaultAppId')}`?